src/App.js

```diff
@@ -38,35 +38,49 @@ const UserIsAuthenticated = UserAuthWrapper({
  * Expose App base component. Handle all routes
  */
 
+const buildRoutes = (store, onLogPageView, defaultRoute, features, userManager) => {
+
+  const getComponent = component => features.authEnabled ? UserIsAuthenticated(component) : component;
+
+  const router = (
+    <Router history={browserHistory} onUpdate={onLogPageView}>
+      <Redirect from="/" to={defaultRoute} />
+      <Route path="login" component={Login} />
+      <Route path="about" component={About} />
+      <Route path="/callback" component={CallbackPage} />
+      {features.trust !== false ?
+        <div>
+          <Route path="search-creator" component={SearchCreator} />
+          <Route path="saved-searches" component={SeeAllSearches}/>
+          <Route path="saved-search/:id" component={SearchDetail} />
+          <Route path="edit-search/:id" component={SearchEditor} />
+        </div>
+      : null}
+      {features.ask ?
+        <div>
+          <Route path="forms" component={getComponent(FormList)} />
```

Comment:
I'd suggest adding something like a wrapper that would work as a middleware in this case `LayoutContainer` decides what renders. 

``` javascript
<Route component={LayoutContainer}>
  <Route path="search-creator" component={SearchCreator} />
  <Route path="saved-searches" component={SeeAllSearches}/>
  <Route path="saved-search/:id" component={SearchDetail} />
  <Route path="edit-search/:id" component={SearchEditor} />
</Route>
```

Also, using something like this you can check auth in each mount and have full control about auth flow
```javascript
class LayoutContainer extends Component {
  componentWillMount () {
    const { dispatch } = this.props;
    dispatch(checkLogin());
  }

  render () {
    const {
      user,
      loadingUser,
      userNotFound
    } = this.props;

    if (userNotFound) return <NotFoundLayout { ...this.props } />;
    if (!user && loadingUser) return <FullLoading />;
    if (user) return <Layout { ...this.props } />;
    return <FullLoading />;
  }
}
```

IMO this way its easier to toggle stuff off. And I find this pattern super confortable.